src/tile/base.ts

```diff
@@ -19,12 +21,14 @@ abstract class BaseTile {
    * Constructor
    * @param url URL for terrain RGB raster tilesets
    * @param tileSize size of tile. 256 or 512
+   * @param tms whether it is Tile Map Service
    * @param minzoom minzoom for terrain RGB raster tilesets
    * @param maxzoom maxzoom for terrain RGB raster tilesets
    */
-  constructor(url: string, tileSize: number, minzoom: number, maxzoom: number) {
+  constructor(url: string, tileSize: number, tms: boolean, minzoom: number, maxzoom: number) {
```

Comment:
Can you move tms option to the last argument of constructor? And please set false as default. So it will not affect existing code